internal/archiver/archiver.go

```diff
@@ -508,6 +513,13 @@ func (arch *Archiver) save(ctx context.Context, snPath, target string, previous
 	case fi.Mode.IsRegular():
 		debug.Log("  %v regular file", target)
 
+		// check if repository size is about to be exceeded.
```

Comment:
This can only check after the fact, but not beforehand. Reading from the summary also requires taking the `arch.mu` mutex.

But there's a larger problem here. When someone backs up very large files, this can miss the target size by hundreds of gigabytes (assuming such large files).